utility/resource_loader_compat.cpp

```diff
@@ -0,0 +1,2096 @@
+	Error err;
+	// We never load scripts, because they are likely incompatible and it's a security issue
+	if (external_resources[i].type.find("Script") != -1){
+		ERR_FAIL_COND_V_MSG(!no_abort_on_ext_load_fail, ERR_CANT_ACQUIRE_RESOURCE, "Refusing to load external script " + external_resources[i].path);
+		set_dummy_ext(i);
+		return OK;
+	}
+
+	external_resources.write[i].cache = rl.load(external_resources[i].path, project_dir, &err);
+	if (external_resources.write[i].cache.is_null()){
+		if (!no_abort_on_ext_load_fail) {
+			ERR_FAIL_COND_V_MSG(err != OK, err, "Failed to load external resource " + external_resources[i].path);
+		} else {
+			if (external_resources.write[i].cache.is_null()){
+				// just do a dummy resource and hope this works anyway
+				set_dummy_ext(i);
+			}
+		}
+	}
+
+	return OK;
+}
+
+RES ResourceLoaderBinaryCompat::get_external_resource(const int subindex){
+	if (external_resources[subindex - 1].cache.is_valid()){
+		return external_resources[subindex - 1].cache;
+	}
+	// We don't do multithreading, so if this external resource is not cached (either dummy or real)
+	// then we return a blank resource
+	return RES();
+}
+
+RES ResourceLoaderBinaryCompat::get_external_resource(const String &path){
+	for (int i = 0; i < external_resources.size(); i++){
+		if (external_resources[i].path == path){
+			return external_resources[i].cache;
+		}
+	}
+	// We don't do multithreading, so if this external resource is not cached (either dummy or real)
+	// then we return a blank resource
+	return RES();
+}
+
+bool ResourceLoaderBinaryCompat::has_external_resource(const String &path)
+{
+	for (int i = 0; i < external_resources.size(); i++){
+		if (external_resources[i].path == path){
+			return true;
+		}
+	}
+	return false;
+}
+
+
+// by default, we don't instance an internal resource.  
+// This is done for compatibility reasons. Class names and constructors, and their properties, have changed between versions.
+// So we instead just make a dummy resource and keep a list of properties, which would all be Variants
+RES ResourceLoaderBinaryCompat::instance_internal_resource(const String &path, const String &type, const int subindex){
+	RES res;
+	// if this is a fake load, create a dummy instead
+	if (fake_load){
+		res = make_dummy(path, type, subindex);
+		return res;
+	}
+	// If this is a real load...
+
+	// TODO: rename godot v2 resources to godot v3 resources
+	// we don't populate the cache by default, so we likely won't hit this (?)
+	if (cache_mode == ResourceFormatLoader::CACHE_MODE_REPLACE && ResourceCache::has(path)) {
+		//use the existing one
+		Resource *r = ResourceCache::get(path);
+		if (r->get_class() == type) {
+			r->reset_state();
+			res = Ref<Resource>(r);
+		}
+	}
+
+	if (res.is_null()) {
+		//did not replace
+		Object *obj = ClassDB::instance(type);
+		if (!obj) {
+			error = ERR_FILE_CORRUPT;
+			ERR_FAIL_V_MSG(ERR_FILE_CORRUPT, local_path + ":Resource of unrecognized type in file: " + type + ".");
```

Comment:
```suggestion
			ERR_FAIL_V_MSG(RES(), local_path + ":Resource of unrecognized type in file: " + type + ".");
```

Return type is `RES` not `Error`.